src/java.base/share/classes/sun/net/www/protocol/jmod/Handler.java

```diff
@@ -43,7 +46,8 @@ protected URLConnection openConnection(URL url) throws IOException {
         String s = url.toString();
         int index = s.indexOf("!/");
         if (index == -1)
-            throw new MalformedURLException("no !/ found in url spec:" + s);
+            throw new MalformedURLException(
+                formatMsg("no !/ found in url spec%s", filterJarName(s).prefixWith(": ")));
```

Comment:
JMOD files can only be used at compile time and link time. So I think you can drop the changes jmod stream handler.